types/market.go

```diff
@@ -41,10 +48,69 @@ var MarketNames = map[MarketID]string{
 	MarketTotalGoalsAway:   "total_goals_away",
 }
 
-type Market interface {
+type Market struct {
+	MarketInterface
+}
+
+type MarketInterface interface {
 	transaction.TransactionMarshaller
 }
 
+func (m *Market) UnmarshalJSON(b []byte) error {
+	json, err := simplejson.NewJson(b)
+	if err != nil {
+		return err
+	}
+
+	marketName, err := json.GetIndex(0).String()
```

Comment:
maybe GetPath would be better?  
https://godoc.org/github.com/bitly/go-simplejson#Json.GetPath

Reply:
or simple Get()